### PR TITLE
Fix AdaShift NaN on the first update with keep_num > 1

### PR DESCRIPTION
**Pull request by winklemad**
### Problem

`AdaShift` NaNs on its first update whenever `keep_num > 1` (the **default is 10**):

```python
import torch
from pytorch_optimizer import AdaShift
p = torch.nn.Parameter(torch.rand(4))
opt = AdaShift([p], lr=1e-2)  # default keep_num=10
for _ in range(20):
    opt.zero_grad(); ((p - 1.0) ** 2).sum().backward(); opt.step()
print(p)  # NaN on main (around the 9th step)
```

`init_group` seeds the gradient queue with the first gradient (`deque([grad.clone()], maxlen=keep_num)`), and `step()` then appends the gradient again — so the queue fills **one step early**. The first update runs at internal step `keep_num - 1`, where

```python
bias_correction = debias(beta2, step - keep_num)  # debias(beta2, -1) -> 1 - beta2**-1 < 0
```

is **negative**. The update is `update.div_(exp_avg_sq.div(bias_correction).sqrt_()...)`, so dividing by a negative and taking the square root yields `sqrt(negative) = NaN`, which permanently corrupts the parameter.

The existing tests only use `keep_num=1` (where the `maxlen=1` queue stays at length 1 and never fills early), so this was never exercised.

### Fix

Initialize the queue empty (`deque(maxlen=keep_num)`) so it fills at the correct step and the first update no longer uses a negative bias correction. Added `test_adashift_default_keep_num_does_not_nan` (NaN on main, finite after the fix); the existing `keep_num=1` tests still pass, `ruff` clean.

### CODE REVIEW

**Review by kozistr — DISMISSED** [review]
thanks! lgtm! could you resolve the conflicts?

**Comment by winklemad** [comment]
Thanks @kozistr! Rebased on main and resolved the conflict (it was just the shared test file — kept both the adasmooth and adashift regression tests). Ready for merge.
